.github/workflows/changed_file_trigger_examples_check.yml

```diff
@@ -0,0 +1,117 @@
+name: change_examples_check_all
+run-name: ${{ github.actor }} is testing out GitHub Actions 🚀
+on:
+  pull_request:
+
+  schedule:
+    # run at 00:00 of every Sunday
+    - cron:  '0 0 * * *'
+
+jobs:
+
+# This is for changed files detect and check.
+  detect-changed-files:
+    runs-on: ubuntu-latest
+    outputs:
+      changedFiles: ${{ steps.changed-files.outputs.all_changed_files }}
+
+    name: Check out all files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+
+      - name: Get all changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          since_last_remote_commit: true  # Using this can trigger action each time a PR is submitted.
+
+
+  deploy:
+    name: Test each changed example files
+    needs: detect-changed-files
+    runs-on: self-hosted
+    container:
+      image: hpcaitech/pytorch-cuda:1.11.0-11.3.0
+      options: --gpus all --rm -v /data/scratch/cifar-10:/data/scratch/cifar-10
+    steps:
+
+      - name: List all changed example files
+        run: |
+          changedFileName=""
+          for file in ${{ needs.detect-changed-files.outputs.changedFiles  }}; do
+            changedFileName="${file}:${changedFileName}"
+          done
+          echo "$changedFileName was changed"
+          res=`python .github/workflows/changed_file.py --fileNameList $changedFileName`
+          echo "$res is res"
+          if [ -z "$res" ]
+          then
+                echo "\No change in example folder!"
+          else
+                pip install -r ./requirements/requirements.txt
+                pip install -e .
+                bash .github/workflows/changed_file.sh "$res"
+          fi
+
+
+
+
+# This is for all files weekly check
+
+  matrix_preparation:
+    name: Prepare Direction List for All files
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+
+    - name: 📚 Checkout
+      uses: actions/checkout@v3
+    - name: setup matrix
+      id: set-matrix
+      run: |
+          dir=`ls ${PWD}/examples`
+          allDir=""
+          for i in ${dir}; do
+            allDir="${i}:${allDir}"
+            echo "${allDir} is allDir"
+          done
+          res=`python .github/workflows/weekly_check.py --fileNameList $allDir`
+          container=$( IFS=',' ; echo "${res[*]}" )
+          echo "$container"
+          echo "::set-output name=matrix::{\"container\":$(echo "$container")}"
+
+
+  weekly_check:
+    if: github.event.schedule == '0 0 * * *'
```

Comment:
This is error-prone check , as we need to keep track of the schedule cron time in both the event trigger and the step condition.

Look at this [discussion](https://stackoverflow.com/questions/70281670/checking-condition-when-step-is-running-on-schedule-in-github-actions) for reference.